lib/public/AppFramework/Db/Repository.php

```diff
@@ -0,0 +1,434 @@
+		}
+
+
+		$update->executeStatement();
+		return $entity;
+	}
+
+	public function delete(object $entity): void {
+		$delete = $this->connection->getQueryBuilder();
+		$delete->delete($this->tableName);
+
+		$foundId = false;
+		foreach ($this->reflection->getProperties() as $property) {
+			$columns = $property->getAttributes(Column::class, \ReflectionAttribute::IS_INSTANCEOF);
+			if (empty($columns)) {
+				continue; // Not in the DB
+			}
+
+			$column = $columns[0]->newInstance();
+
+			if (count($property->getAttributes(Id::class, \ReflectionAttribute::IS_INSTANCEOF)) !== 0) {
+				$delete->andWhere($delete->expr()->eq($column->name, $property->getValue($entity)));
+				$foundId = true;
+			};
+		}
+
+		if (!$foundId) {
+			throw new \LogicException("The given entity is missing a required #[Id] attribute on one of its properties.");
+		}
+
+		$delete->executeStatement();
+	}
+
+	/**
+	 * @param Types::* $type
+	 * @return IQueryBuilder::PARAM_*
+	 */
+	private function getParameterType(string $type, bool $isArray): string|int {
+		if ($isArray) {
+			return match ($type) {
+				Types::INTEGER, Types::SMALLINT => IQueryBuilder::PARAM_INT_ARRAY,
+				Types::STRING => IQueryBuilder::PARAM_STR_ARRAY,
+				Types::JSON => IQueryBuilder::PARAM_JSON,
+				default => throw new \LogicException("Parameter type '$type' is not supported as an array."),
+			};
+		}
+
+		return match ($type) {
+			Types::INTEGER, Types::SMALLINT => IQueryBuilder::PARAM_INT,
+			Types::STRING => IQueryBuilder::PARAM_STR,
+			Types::BOOLEAN => IQueryBuilder::PARAM_BOOL,
+			Types::BLOB => IQueryBuilder::PARAM_LOB,
+			Types::DATE, Types::DATETIME => IQueryBuilder::PARAM_DATETIME_MUTABLE,
+			Types::DATETIME_TZ => IQueryBuilder::PARAM_DATETIME_TZ_MUTABLE,
+			Types::DATE_IMMUTABLE => IQueryBuilder::PARAM_DATE_IMMUTABLE,
+			Types::DATETIME_IMMUTABLE => IQueryBuilder::PARAM_DATETIME_IMMUTABLE,
+			Types::DATETIME_TZ_IMMUTABLE => IQueryBuilder::PARAM_DATETIME_TZ_IMMUTABLE,
+			Types::TIME => IQueryBuilder::PARAM_TIME_MUTABLE,
+			Types::TIME_IMMUTABLE => IQueryBuilder::PARAM_TIME_IMMUTABLE,
+			Types::JSON => IQueryBuilder::PARAM_JSON,
+			default => IQueryBuilder::PARAM_STR,
+		};
+	}
+
+	/**
+	 * Finds entities by a set of criteria.
+	 *
+	 * Use the property names for the criteria and orderBy key.
+	 *
+	 * @param array<string, int|float|string> $criteria
+	 * @param array<string, 'asc'|'desc'>|null $orderBy
+	 * @return \Generator<T>
+	 * @since 33.0.0
+	 */
+	public function findBy(array $criteria, array $orderBy = [], int|null $limit = null, int|null $offset = null): \Generator {
+		$qb = $this->getSelectQueryBuilder($criteria, $orderBy);
+
+		if ($limit !== null) {
+			$qb->setMaxResults($limit);
+		}
+
+		if ($offset !== null) {
+			$qb->setFirstResult($offset);
+		}
+
+		return $this->yieldEntities($qb);
+	}
+
+	public function deleteBy(array $criteria, int|null $limit = null): void {
```

Comment:
Should return the result of executeStatement 
```suggestion
	public function deleteBy(array $criteria, int|null $limit = null): int {
```